provider/dnsimple.go

```diff
@@ -25,67 +25,69 @@ import (
 
 	"github.com/dnsimple/dnsimple-go/dnsimple"
 	log "github.com/sirupsen/logrus"
+	"golang.org/x/oauth2"
 
 	"sigs.k8s.io/external-dns/endpoint"
+	"sigs.k8s.io/external-dns/pkg/apis/externaldns"
 	"sigs.k8s.io/external-dns/plan"
 )
 
 const dnsimpleRecordTTL = 3600 // Default TTL of 1 hour if not set (DNSimple's default)
 
-type identityService struct {
+type dnsimpleIdentityService struct {
 	service *dnsimple.IdentityService
 }
 
-func (i identityService) Whoami() (*dnsimple.WhoamiResponse, error) {
-	return i.service.Whoami()
+func (i dnsimpleIdentityService) Whoami() (*dnsimple.WhoamiResponse, error) {
+	return i.service.Whoami(context.TODO())
 }
 
-// Returns the account ID given dnsimple credentials
-func (p *dnsimpleProvider) GetAccountID(credentials dnsimple.Credentials, client dnsimple.Client) (accountID string, err error) {
-	// get DNSimple client accountID
-	whoamiResponse, err := client.Identity.Whoami()
-	if err != nil {
-		return "", err
-	}
-	return strconv.Itoa(whoamiResponse.Data.Account.ID), nil
-}
-
-// dnsimpleZoneServiceInterface is an interface that contains all necessary zone services from dnsimple
+// dnsimpleZoneServiceInterface is an interface that contains all necessary zone services from DNSimple
 type dnsimpleZoneServiceInterface interface {
 	ListZones(accountID string, options *dnsimple.ZoneListOptions) (*dnsimple.ZonesResponse, error)
```

Comment:
can we drop this same as ListRecords? seems to be the same method as without context
you might also can cleanup the test file

Reply:
Just to clarify, do you want me to remove the non-context `ListRecords` and `ListZones` methods?

In case, should I actually drop the `WithContext` versions (`ListRecordsWithContext`) and simply add the context in the existing method signature?

Reply:
Yes this would be perfect 👍 , just use the context in the existing method signature 🙂 .

Reply:
Changes made.